Data Grid/Prometheus/Grafana quickstart demo

vblagoje opened the pull request:
See README for details. @galderz this is a preview of a quickstart, would test.sh be relevant for this quickstart demo where everything is run from the command line? If so, should I perhaps add checks that demo deploys succesfully and to verify that certain routes and services exist at the end of the demo?

galderz committed:
JDG-2259 OpenShift tutorials

* Shared 'Hello World' tutorial
* Custom XML tutorial. Also shows how to disable auth and encrypt.
* Cross-site tutorial.
* External access tutorial over Hot Rod and HTTPs.
* Create cache tutorial

oraNod committed:
JDG-2280 Add OpenShift tutorial READMEs

* Create caches for cache and data grid.
* External access.
* User configuration.
* Cross-site.

galderz committed:
JDG-2259 Update OpenShift tutorials for 7.3

* Temporarily use community client version.

Commit:
Add Prometheus Data Grid quickstart

galderz commented on openshift/datagrid-service/prometheus-datagrid/1-create-project-add-permissions.sh:
```diff
@@ -0,0 +1,10 @@
+#!/bin/bash
+
+
+oc login -u system:admin
```
If you need to logging as system administrator, you should make a note in the README file that the `minishift` profile needs to have admin-user enabled, e.g.

```
minishift addon enable admin-user
```

Also, AFAIK that requires recreating the VM, which should be noted.

galderz commented on openshift/datagrid-service/prometheus-datagrid/README.md:
```diff
@@ -0,0 +1,96 @@
+
+About This Quickstart
+---------------------
+This quickstart demonstrates how to integrate Data Grid with Prometheus and Grafana
```
Are all these steps required in all supported OpenShift versions? Are there any OpenShift versions, e.g. `3.11` or `4` where some of these steps can be skipped?

galderz commented on openshift/datagrid-service/prometheus-datagrid/3-install-datagrid.sh:
```diff
@@ -0,0 +1,24 @@
+
+echo "Installing RHDG cluster"
+
+oc create -f datagrid73-basic.json
```
~Will this work at all? There's no `datagrid73-basic.json` file in this repo...~

I see you added a template to the repo. That should not happen.

galderz replied:
Also, `datagrid73-basic.json` are templates no longer in use. You should use datagrid-service templates, e.g. [this](https://github.com/jboss-container-images/jboss-datagrid-7-openshift-image/blob/datagrid73-dev/services/datagrid-service-template.yaml)...

galderz commented on openshift/datagrid-service/prometheus-datagrid/3-install-datagrid.sh:
```diff
@@ -0,0 +1,24 @@
+RHDG_PASSWORD=developer
+
+oc new-app datagrid73-basic \
+  -p USERNAME=$RHDG_USER \
```
These are not valid parameters in service templates.

galderz reviewed (changes requested):
Several comments. Also, there should be no templates included in the tutorial. Creates duplication and extra maintenance work :(

Yes, we do need a test.sh that automates all the installation steps and verifies that in the end result the correct output or stats are returned by the corresponding URL.

Finally, if you're adding screenshots, it'd be nice to have a screenshot of the metrics at the end.

vblagoje commented:
@galderz I updated quickstart to use datagrid-service-template.yaml, I updated instructions, screenshots and added test.sh. However, I need to consult with you about test.sh a bit more. I would keep it as-is and then use some type of a check at the end to make sure that pods are running and that routes are open. Something like that. Not sure how to check output of oc commands but IIRC I've seen other do checks with some type of json path value boolean expressions from oc command responses.... wdyt?

vblagoje commented on openshift/datagrid-service/prometheus-datagrid/3-install-datagrid.sh:
```diff
@@ -0,0 +1,25 @@
+
+echo "Installing RHDG cluster"
+
+oc create -f https://raw.githubusercontent.com/jboss-container-images/jboss-datagrid-7-openshift-image/datagrid73-dev/services/datagrid-service-template.yaml
```
@galderz here is the change. I used datagrid73-dev branch and datagrid-service-template for template

vblagoje commented on openshift/datagrid-service/prometheus-datagrid/3-install-datagrid.sh:
```diff
@@ -0,0 +1,25 @@
+  -p APPLICATION_USER=developer \
+  -p APPLICATION_PASSWORD=developer \
+  -e AB_PROMETHEUS_ENABLE=true \
+  -p IMAGE=brew-pulp-docker01.web.prod.ext.phx2.redhat.com:8888/jboss-datagrid-7/datagrid73-openshift:jb-datagrid-7.3-openshift-dev-rhel-7-containers-candidate-46261-20181219020350
```
@galderz I also used a working image here until we have an official one

vblagoje commented:
Update with your feedback @galderz (I forced pushed the change apologies). The official template is used and a working image

Commit:
Remove template copy, use the main/original template

vblagoje commented on openshift/datagrid-service/prometheus-datagrid/3-install-datagrid.sh:
```diff
@@ -0,0 +1,25 @@
+  -p APPLICATION_USER=developer \
+  -p APPLICATION_PASSWORD=developer \
+  -e AB_PROMETHEUS_ENABLE=true \
+  -p IMAGE=brew-pulp-docker01.web.prod.ext.phx2.redhat.com:8888/jboss-datagrid-7/datagrid73-openshift:jb-datagrid-7.3-openshift-dev-rhel-7-containers-candidate-46261-20181219020350
```
@oraNod here are the new params

galderz replied:
The brew image link should not really be part of the tutorial

galderz replied:
IOW, leave it out

slaskawi commented:
There's an OpenShift 4 demo environment available on RHPDS. You also may use it to setup everything :)

galderz commented:
Superseded by #257
